internals-js/src/federation.ts

```diff
@@ -563,7 +594,7 @@ export class FederationMetadata {
   }
 
   allFederationDirectives(): DirectiveDefinition[] {
-    const baseDirectives = [
+    const baseDirectives: DirectiveDefinition<any>[] = [
```

Comment:
Nit: we can save the `<any>` in `DirectiveDefinition<any>[]` if we want (it avoid a `any` and would more consistent with the method signature too).